(Back)port of PAXEXAM-926 (RMI port range is not configurable for forked container) towards the v4.x branch

glimmerveen opened the pull request:
I've prepared a (back)port of an enhancement that was previously merged towards master (PAXEXAM-926). This enhancement is about being able to configure the RMI port (range) of a forked container. 

This is particular useful features when running tests in parallel through surefire/failsafe's fork feature:

```xml
<configuration>
    <systemPropertyVariables>
        <pax.exam.forked.invoker.port.range.lowerbound>1${surefire.forkNumber}000</pax.exam.forked.invoker.port.range.lowerbound>
        <pax.exam.forked.invoker.port.range.upperbound>1${surefire.forkNumber}009</pax.exam.forked.invoker.port.range.upperbound>
    </systemPropertyVariables>
    <forkCount>2</forkCount>
</configuration>
```

Without this enhancement, the parallel execution of the tests in most cases fail due to multiple tests trying to claim the same RMI port at the same time.

The original commits used (by cherrie-picking):

* a6270b9 by @nfalco79 which was the initial commit of this feature (PAXEXAM-926)
* 2b4a85e by @rmannibucau which contained some improvements related to the unit tests of this feature.

nfalco79 committed:
This is a (back)port of an enhancement merged to master back in 2019. Cherry-pick of commit a6270b9 (enhancement tracked before as PAXEXAM-926, and merged through #79).

rmannibucau committed:
This is a (back)port of an update to an enhancement merged to master back in 2019. The update was, at least according to the associated PR, primarily focused on Karaf embedded runner, but there were some (test) changes in the pax-exam-container-forked as well.
Cherry-pick of commit 2b4a85e (merged through #89).

oliverlietz commented:
I prefer to accept an optional single port for RMI and use a random free port if no port was provided.

glimmerveen committed:
Removed the constants and the initialization of the exam invoker port. It was referenced within the ForkedTestContainer, but nothing was actually tied to that port.

Now only the ForkedFrameworkFactory deals with RMI port, which make sense as it is that class that actually manages a RMI LocateRegistry.

glimmerveen commented:
> I prefer to accept an optional single port for RMI and use a random free port if no port was provided.

Hi Oliver, thanks for looking into this PR. I must admit I was initially focussed on porting the existing functionality, without really comprehending what all the elements within this PR were actually doing.

After having another look at it, I noticed that the port determined within the ForkedTestContainer class was not actually being used, so I opted for removing that from the PR with commit ce5d647.

This should already simplify this PR a bit. I see some benefit of the flexibility of having a range specified that from which the ForkedFrameworkFactory can find a free port, ie being able to deal with preceding's test RMI port not being released fully by the OS. But I'll admit I haven't seen this actually go wrong in this case, so that may be a matter of over-engineering present in the original solution.

If you prefer I can remove the range portion of the configuration, and only have the plain `pax.exam.forked.invoker.port` as a means to control the RMI port being used.

oliverlietz commented:
@glimmerveen, Thanks for driving this topic! Please see #1128.
